include/solvers/navier_stokes_base.h

```diff
@@ -298,6 +298,12 @@ class NavierStokesBase : public PhysicsSolver<VectorType>
   virtual void
   update_multiphysics_time_average_solution() = 0;
 
+  /**
+   * @brief  Required only for the matrix-free solver to update solutions within iterate function.
+   */
+  virtual void
+  update_solutions_for_multiphysics() = 0;
```

Comment:
Would it be not better to have a default version of this virtual functoin that is not a pure virtual function but that does nothing and then just override it in the matrix free?
This would be less invasive I think

Reply:
Yes, this makes a lot of sense. I will do the changes.